src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```
@@ -291,5 +293,22 @@ private async Task UpdateJobAsync(CancellationToken cancellationToken)
                 _weakETag = wrapper.ETag;
             }
         }
+
+        private List<string> GetSearchParamHashesForResources(List<string> resourceList)
+        {
+            List<string> searchParamHashes = new List<string>();
+            foreach (string resource in resourceList)
+            {
+                if (!_reindexJobRecord.ResourceTypeSearchParameterHashMap.ContainsKey(resource))
+                {
+                    _logger.LogError($"Unable to find search parameter hash for resource type {resource}");
+                    throw new Exception();
```

This should throw an exception with more information.